src/ADAL.PCL/Internal/Telemetry/HttpEvent.cs

```diff
@@ -0,0 +1,117 @@
+﻿//----------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Text;
+
+namespace Microsoft.IdentityModel.Clients.ActiveDirectory
+{
+    internal class HttpEvent : DefaultEvent
+    {
+        internal string UserAgent { get; set; }
+
+        internal string HttpMethod { get; set; }
+
+        internal HttpStatusCode HttpResponseCode { get; set; }
+
+        internal string HttpQueryParameters { get; set; }
+
+        internal string HttpResponseMethod { get; set; }
+
+        internal string ResponseApiVersion { get; set; }
+
+        internal void ParseQuery(string Query)
```

Comment:
As mentioned in the earlier comment, they are not duplicates. There is difference in the parsing logic.